tests/components/radarr/test_init.py

```diff
@@ -62,3 +69,24 @@ async def test_device_info(
     assert device.manufacturer == DEFAULT_NAME
     assert device.name == "Mock Title"
     assert device.sw_version == "10.0.0.34882"
+
+
+@pytest.mark.freeze_time("2021-12-03 00:00:00+00:00")
+async def test_update_failed(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test coordinator updates handle failures."""
+    entry = await setup_integration(hass, aioclient_mock)
+    assert entry.state is ConfigEntryState.LOADED
+    entity = "sensor.mock_title_disk_space_downloads"
+    assert hass.states.get(entity).state == "263.10"
+
+    with patch(
+        "homeassistant.components.radarr.RadarrClient._async_request",
+        side_effect=ArrConnectionException,
+    ) as updater:
+        next_update = dt_util.utcnow() + timedelta(minutes=1)
+        async_fire_time_changed(hass, next_update)
+        await hass.async_block_till_done()
+        assert updater.call_count == 2
+        assert hass.states.get(entity).state == STATE_UNAVAILABLE
```

Comment:
For completeness perhaps it could be worth removing the side effect and recover to see if state comes back as it should